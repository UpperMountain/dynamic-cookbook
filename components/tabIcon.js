import React from "react";
import { MaterialIcons } from "@expo/vector-icons";

export default function tabIcon(name) {
  return function TabIcon({ tintColor }) {
    return <MaterialIcons name={name} color={tintColor} size={24} />;
  };
}
