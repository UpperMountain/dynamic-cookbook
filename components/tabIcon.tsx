import React from "react";
import { MaterialIcons, MaterialCommunityIcons } from "@expo/vector-icons";

interface Props {
  tintColor?: string;
}

export default function tabIcon(name: string) {
  return function TabIcon({ tintColor }: Props) {
    return <MaterialIcons name={name} color={tintColor} size={24} />;
  };
}

export function communityTabIcon(name: string) {
  return function TabIcon({ tintColor }: Props) {
    return <MaterialCommunityIcons name={name} color={tintColor} size={24} />;
  };
}
