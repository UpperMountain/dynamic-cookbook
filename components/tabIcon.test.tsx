import React from "react";
import tabIcon from "./tabIcon";
import renderer from "react-test-renderer";

describe('tabIcon("home")', () => {
  const HomeIcon = tabIcon("home");
  it("should render without exploding", () => {
    renderer.create(<HomeIcon />).unmount();
  });

  it("should pass snapshot tests", () => {
    const cmp = renderer.create(
      <>
        <HomeIcon />
        <HomeIcon tintColor="red" />
      </>
    );
    expect(cmp.toJSON()).toMatchSnapshot();
    cmp.unmount();
  });
});
