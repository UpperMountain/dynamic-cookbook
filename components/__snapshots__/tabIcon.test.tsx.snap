// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communityTabIcon("account") should pass snapshot tests 1`] = `
Array [
  <Text />,
  <Text />,
]
`;

exports[`tabIcon("home") should pass snapshot tests 1`] = `
Array [
  <Text />,
  <Text />,
]
`;
